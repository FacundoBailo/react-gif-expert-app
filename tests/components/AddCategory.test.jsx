const { render, screen, fireEvent } = require("@testing-library/react");
const { default: AddCategory } = require("../../src/components/AddCategory");


describe('probando <AddCategory/>', () => { 

    const inputValue = 'Goku';


    test('debe hacer match con el snapshot', () => { 

        const { container } = render( <AddCategory onNewCategory={() => {}}/> );

        expect( container ).toMatchSnapshot();

     })

     test('debe cambiar el valor de la caja de texto', () => { 

        render( <AddCategory onNewCategory={() => {}}/> );

        const input = screen.getByRole('textbox');

        fireEvent.input( input, { target: { value: inputValue } } )

        expect( input.value ).toBe('Goku');

      })

    test('debe de llamar onNewCategory si el input tiene un valor', () => { 

        render( <AddCategory onNewCategory={() => {}}/> );

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: inputValue } } )
        fireEvent.submit( form );
        //screen.debug()

        expect( input.value ).toBe('');

    })

 })