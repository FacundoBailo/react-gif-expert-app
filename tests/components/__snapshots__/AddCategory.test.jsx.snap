// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <AddCategory/> debe hacer match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
